### PR TITLE
feat(ui): "Remove Site" action wired to SiteStore.remove(id:) (#186)

### PR DESCRIPTION
**Pull request by davidwkeith**
## Summary

Wires the existing `SiteStore.remove(id:)` to a UI action — closes #186. The store method was correct but had **zero UI consumers**, so an owner had no way to forget a site from the app's list. This became load-bearing after #184/#185 made `refresh()` deliberately never auto-prune a bookmarked entry: an explicit user action is now the only correct way to drop a stale/abandoned site.

## Changes

- **`SitesLauncherView`** — each site row gets a right-click **context menu** *and* a trailing **swipe action** ("Remove from Anglesite…"), gated behind a `.confirmationDialog`. Copy makes clear this only forgets the site: *"The files in `<path>` are left on disk."*
- **`removeSite(_:)`** — calls `remove(id:)` then prunes the local list directly rather than re-running `refreshSites()`, which on DevID would instantly rediscover a still-present in-root folder and undo the removal visually.
- **MAS bookmark drop is automatic** — per-site bookmarks live inline in `Site.bookmarkData`, so `remove(id:)` already drops the security-scoped grant; no separate store to clean up.
- **Test** — `removeBookmarkedSitePermanent` proves a bookmarked entry (the #184/#185 case `refresh()` never prunes) stays gone, with its bookmark, across a reload + refresh that can't rediscover it.

## Acceptance criteria

- [x] User can remove a site via the UI; `sites.json` no longer contains it after relaunch
- [x] Files under the site folder remain on disk
- [x] (MAS) the entry's bookmark is dropped on removal
- [x] Test coverage for the action path

## Scope notes

- Skipped the optional focused-`SiteWindow` menu command (the issue lists it as "and/or"; extra FocusedValue plumbing for no added coverage).
- Inherent, consistent with the issue: on **DevID**, removing a site whose folder still lives under `~/Sites` reappears on the next rescan — removal is durable for out-of-root and bookmarked-but-deleted sites (the target case). `refresh()` retention is left untouched per the issue's out-of-scope note.

## Testing

- `swift test --filter SiteStoreTests` → 19/19 pass
- Both `Anglesite` (DevID) and `AnglesiteMAS` schemes build clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Sources/AnglesiteApp/SitesLauncherView.swift [hunk 1]
```diff
@@ -125,8 +128,34 @@ struct SitesLauncherView: View {
...
+            }
+            .swipeActions(edge: .trailing) {
+                Button("Remove", systemImage: "minus.circle", role: .destructive) {
+                    siteToRemove = site
+                }
+            }
         }
         .listStyle(.inset)
+        .confirmationDialog(
+            "Remove “\(siteToRemove?.name ?? "")” from Anglesite?",
```
Comment (davidwkeith) [inline Sources/AnglesiteApp/SitesLauncherView.swift]: The title string reads from `siteToRemove?.name` directly, so it evaluates to `"Remove "" from Anglesite?"` once `siteToRemove` is nil'd during dismiss. SwiftUI may capture the last-rendered state for the outgoing animation, so this flash might not be visible in practice — but the dependency on mutable state in the title string is fragile.

A safer pattern is to keep a separate non-optional for the in-flight name:

```swift
@State private var siteToRemoveName = ""

// in context menu / swipe action:
siteToRemove = site
siteToRemoveName = site.name

// title:
"Remove "\(siteToRemoveName)" from Anglesite?"
```

That way the title stays stable across the full dismiss animation regardless of when `siteToRemove` is cleared.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Sources/AnglesiteApp/SitesLauncherView.swift [hunk 2]
```diff
@@ -180,6 +209,23 @@ struct SitesLauncherView: View {
         dismissWindow()
     }
 
+    /// Forget `site` from the registry without touching its files. On MAS this also drops the
+    /// site's persisted security-scoped bookmark, since that lives inline in the `Site` entry.
+    /// We prune the local list directly rather than re-running `refreshSites()`: a DevID rescan
+    /// of `~/Sites` would immediately rediscover a still-present in-root folder, undoing the
+    /// removal visually. Persistence is handled by `remove(id:)`.
+    private func removeSite(_ site: SiteStore.Site) {
+        siteToRemove = nil
```
Comment (davidwkeith) [inline Sources/AnglesiteApp/SitesLauncherView.swift]: SwiftUI auto-dismisses `confirmationDialog` when any button is tapped — it calls the `isPresented` binding's `set` with `false` (which clears `siteToRemove` via the hand-rolled binding) automatically, after the action closure returns. This explicit nil fires synchronously in the action closure before that auto-dismiss fires, making it redundant.

Same applies to the `Button("Cancel", role: .cancel) { siteToRemove = nil }` action on line 153 — the `.cancel` role already causes automatic dismissal.

Both explicit nils can be removed; SwiftUI owns the dismiss lifecycle here.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Sources/AnglesiteApp/SitesLauncherView.swift [hunk 3]
```diff
@@ -180,6 +209,23 @@ struct SitesLauncherView: View {
...
+        Task {
+            do {
+                try await SiteStore.shared.remove(id: site.id)
```
Comment (davidwkeith) [inline Sources/AnglesiteApp/SitesLauncherView.swift:237]: What's the expected UX if the site being removed has an open `SiteWindow`? After `remove(id:)` fires the store's change handler, does `SiteWindow` subscribe to that and react (close itself, show a "this site was removed" banner, etc.), or does it stay open showing a now-orphaned site?

If `SiteWindow` doesn't subscribe to the change handler, the window will keep running the dev server / MCP subprocess for a site that no longer exists in the registry. Worth either documenting the current behavior here or adding a follow-up issue.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by davidwkeith — COMMENTED** [review]
Overall this is solid — the store already had `remove(id:)` wired up correctly, the UI changes are minimal and well-scoped, and the test covers the key edge case (bookmarked out-of-root site surviving a reload + refresh). Three points to address:

1. **Dialog title reads from mutable state** — `siteToRemove?.name` in the title string collapses to `""` once `siteToRemove` is nil'd, which happens synchronously during dismiss. Add a separate `@State var siteToRemoveName: String` that's set alongside `siteToRemove` and never cleared until the next removal. See inline.

2. **Redundant nil-outs** — SwiftUI auto-dismisses `confirmationDialog` (calling the `isPresented` binding's setter) when any button is tapped. The explicit `siteToRemove = nil` at the top of `removeSite` and in the Cancel button action are both redundant. Drop them. See inline.

3. **Open `SiteWindow` lifecycle** — if the site has an open window, that window is not signalled by this change. Whether the store's change handler covers this case should be documented or tracked as a follow-up. See inline.

---
_Generated by [Claude Code](https://claude.ai/code)_
